Cleaning DB/3 Reports cleaning.sql

```diff
@@ -0,0 +1,28 @@
+use CleaningDB 
+go
+
+select DeepCleans = 
+    concat(c.LastName, ', ', c.FirstName, ' (', c.PhoneNum, ') - ', ((c.RatePerHour * 1.5) * (c.AvgTime * 2))) 
+from Customer c
+
+select CustomerInfo = concat(c.LastName, ' - ', c.Frequency), NumOfMonths = datediff(month, c.StartDate, getdate())
+from Customer c 
+where c.EndDate is null
+order by NumOfMonths desc
+
+select AvgRatePerHourRegClean = avg(c.RatePerHour), AvgRatePerHourDeepClean = avg(c.RatePerHour * 1.5)
+from Customer c
+
+select PricePerYear = 
+    case Frequency 
+        when 'weekly' then (c.RatePerHour * 52)
+        else (c.RatePerHour * 26)
+    end, * 
+from Customer c 
+where (c.RatePerHour = 35) and (c.AvgTime = 7) and (c.Frequency = 'weekly')
```

Comment:
This is correct in the case right now but isn't the best way to provide a report as data can change.  INstead of looking up which customer pays the most and then adding a where clause with the information as you did, it would be better to just select the top customer that came up from the result set.  Then even if the data changes, the report will still return the customer that pays the most

Reply:
I changed this, but I left in some of the where clause based on how many hours the house took to clean.

Reply:
I left it in because there are two customers technically paying the same amount per year, but one house takes seven hours and the other one take eight, so I figured I would use the where clause to eliminate the eight-hour house.

Reply:
Got it, that's good then